src/generator_files.rs

```diff
@@ -49,32 +49,34 @@ fn enumerate_files(
     // Get an iterator over the files in the root path
     let supported_extensions = ["jpg", "jpeg", "png", "bmp", "gif", "webp"];
 
-    let files = walkdir::WalkDir::new(&source_config.root_path)
+    // Use streaming walkdir to avoid loading all files into memory at once
+    let _supported_extensions = ["jpg", "jpeg", "png", "bmp", "gif", "webp"];
+    let walker = walkdir::WalkDir::new(&source_config.root_path)
         .follow_links(false)
         .into_iter()
-        .filter_map(|e| e.ok());
-
-    // We need to materialize the file list to be able to shuffle it
-    let mut files_list: Vec<walkdir::DirEntry> = files
+        .filter_map(|e| e.ok())
         .filter_map(|entry| {
             let path = entry.path();
-            let file_name = path.to_string_lossy().into_owned();
+            let file_name = path.to_string_lossy().to_lowercase();
             if supported_extensions
                 .iter()
-                .any(|&ext| file_name.to_lowercase().ends_with(ext))
+                .any(|&ext| file_name.ends_with(ext))
             {
                 Some(entry)
             } else {
                 None
             }
-        })
-        .collect();
+        });
 
-    // If shuffle is set, shuffle the files
-    if source_config.random_sampling {
-        let mut rng = rand::rng(); // Get a random number generator, thread local. We don´t seed, so typically won't be reproducible
-        files_list.shuffle(&mut rng); // This happens in place
-    }
+    // For random sampling, we need to collect files, but limit the collection size
+    let mut files_list: Vec<walkdir::DirEntry> = if source_config.random_sampling {
+        walker
+            .take(limit * 2) // Limit collection size to prevent memory issues
```

Comment:
this could be a setting, full randomness if need be, guess is that this is rarely useful